core/message_pool.go

```diff
@@ -230,12 +263,49 @@ func UpdateMessagePool(ctx context.Context, pool *MessagePool, store *hamt.CborI
 		pool.Remove(cid)
 	}
 
+	// prune all messages that have been in the pool too long
+	return pool.timeoutMessages(ctx, store, new)
```

Comment:
@acruikshank I think you missed the fact that this method mutates `new`, so it's not passed through as expected to this method. I think the tests you added break when this is fixed (e.g. by #2272).